From f68c7c618a3ae1ca1971431c48b24ddd10d479c4 Mon Sep 17 00:00:00 2001
From: Wei Liu <wei.liu2@citrix.com>
Date: Mon, 27 Feb 2017 12:20:26 +0000
Subject: [PATCH] libs/devicemodel: free xencall handle in error path in
 _open()

Change the allocation to use calloc to get zeroed structure. Free
xencall handler in error path.

Spotted by Coverity.

Signed-off-by: Wei Liu <wei.liu2@citrix.com>
Reviewed-by: Paul Durrant <paul.durrant@citrix.com>
Acked-by: Ian Jackson <ian.jackson@eu.citrix.com>
diff --git a/tools/libs/devicemodel/core.c b/tools/libs/devicemodel/core.c
index 19ebef6..a85cb49 100644
--- a/tools/libs/devicemodel/core.c
+++ b/tools/libs/devicemodel/core.c
@@ -24,7 +24,7 @@
 xendevicemodel_handle *xendevicemodel_open(xentoollog_logger *logger,
                                            unsigned open_flags)
 {
-    xendevicemodel_handle *dmod = malloc(sizeof(*dmod));
+    xendevicemodel_handle *dmod = calloc(1, sizeof(*dmod));
     int rc;
 
     if (!dmod)
@@ -54,6 +54,7 @@ xendevicemodel_handle *xendevicemodel_open(xentoollog_logger *logger,
 
 err:
     xtl_logger_destroy(dmod->logger_tofree);
+    xencall_close(dmod->xcall);
     free(dmod);
     return NULL;
 }
